attest/eventlog.go

```diff
@@ -529,20 +530,20 @@ func parseRawEvent2(r *bytes.Buffer) (event rawEvent, err error) {
 	if err := binary.Read(r, binary.LittleEndian, &numDigests); err != nil {
 		return event, err
 	}
+
 	for i := 0; i < int(numDigests); i++ {
 		var algID uint16
 		if err := binary.Read(r, binary.LittleEndian, &algID); err != nil {
 			return event, err
 		}
 		var digest []byte
-		switch algID {
-		case algSHA1:
-			digest = make([]byte, crypto.SHA1.Size())
-		case algSHA256:
-			digest = make([]byte, crypto.SHA256.Size())
-		default:
-			// ignore signatures that aren't SHA1 or SHA256
-			continue
+		for _, alg := range specID.algs {
+			if alg.ID == algID {
+				digest = make([]byte, alg.Size)
```

Comment:
nit: add a check here to make sure this can't allocate a huge amount of memory and add a break

```
if alg.ID != algID {
    continue
}
if len(r.Len()) > alg.Size {
    return event, fmt.Errorf("reading digest: %v", io.ErrUnexpectedEOF)
}
digest = make([]byte, alg.Size)
break
```

Reply:
```
if len(r.Len()) > alg.Size {
```
Do you mean the other way around here?

Reply:
woops, yeah you're right